plugins/mimecast/komand_mimecast/util/api.py

```diff
@@ -82,7 +86,58 @@ def get_remediation_incident(self, data: dict) -> dict:
     def find_remediation_incidents(self, data: dict) -> dict:
         return self._handle_rest_call("POST", f"{API}/ttp/remediation/find-incidents", data=data)
 
-    def _prepare_header(self, uri) -> dict:
+    def get_siem_logs(self, data: Dict[str, Any]) -> Union[List[Dict], Dict]:
+        uri = f"{API}/audit/get-siem-logs"
+        default_data = {
+            "compress": True,
+            "file_format": "JSON",
+            "type": "MTA",
+        }
+        data.update(default_data)
+        payload = {DATA_FIELD: ([convert_dict_to_camel_case(data)] if data is not None else [])}
+
+        try:
+            request = requests.request(
+                method="POST", url=f"{self.url}{uri}", headers=self._prepare_header(uri), data=str(payload)
+            )
+        except requests.exceptions.RequestException as e:
+            raise PluginException(preset=PluginException.Preset.SERVER_ERROR, data=e)
+
+        try:
+            response = request.json()
+        except json.decoder.JSONDecodeError:
+            combined_json_list = self._handle_zip_file(request)
+            return combined_json_list, request.headers
+
+        if response.get(FAIL_FIELD):
+            self._handle_error_response(response)
+
+        status_code = response.get(META_FIELD).get(STATUS_FIELD)
+        self._handle_status_code_response(response, status_code)
+
+    def _handle_status_code_response(self, response: requests.request, status_code: int):
+        if status_code == 403:
+            raise PluginException(preset=PluginException.Preset.API_KEY, data=response)
+        elif status_code == 404:
+            raise PluginException(preset=PluginException.Preset.NOT_FOUND, data=response)
+        elif status_code >= 500:
+            raise PluginException(preset=PluginException.Preset.SERVER_ERROR, data=response)
+        else:
+            self.logger.error(response)
+            raise PluginException(
+                cause="Server request failed.",
+                assistance=f"Status code is {status_code}, see log for details.",
+                data=response.get(FAIL_FIELD),
+            )
+
+    def _handle_zip_file(self, request: requests.request) -> List[Dict]:
```

Comment:
Docstring for this function please.
The code is quite complex so it would be good to have a description for it